esp/esdllib/esdl_script.cpp

```diff
@@ -769,155 +1046,145 @@ class CEsdlCustomTransform : public CInterfaceOf<IEsdlCustomTransform>
+        xpathContext->addInputValue("espUserName", "");
+        xpathContext->addInputValue("espUserRealm", "");
+        xpathContext->addInputValue("espUserPeer", "");
+        xpathContext->addInputValue("espUserStatus", "");
+        xpathContext->addInputValue("espUserStatusString", "");
+    }
 
-        //external parameters need <es:param> statements to make them accessible (in strict mode)
-        Owned<IPropertyTreeIterator> configParams;
-        if (bindingCfg)
-            configParams.setown(bindingCfg->getElements("Transform/Param"));
-        if (configParams)
-        {
-            ForEach(*configParams)
-            {
-                IPropertyTree & currentParam = configParams->query();
-                if (currentParam.hasProp("@select"))
-                    xpathContext->addInputXpath(currentParam.queryProp("@name"), currentParam.queryProp("@select"));
-                else
-                    xpathContext->addInputValue(currentParam.queryProp("@name"), currentParam.queryProp("@value"));
-            }
-        }
-        if (!strictParams)
-            xpathContext->declareRemainingInputs();
+    //the need for this will go away when the right hand side is libxml2 based
+    StringBuffer content;
+    scriptCtx->toXML(content, srcSection, false);
 
-        Owned<IPropertyTree> theroot = createPTreeFromXMLString(content.str());
-        StringBuffer defaultTarget;
-            //This default gives us backward compatibility with only being able to write to the actual request
-        const char *tgtQueryName = tgtcfg->queryProp("@queryname");
-        defaultTarget.setf("soap:Body/%s/%s", tgtQueryName ? tgtQueryName : method, reqtype);
+    Owned<IPropertyTree> theroot = createPTreeFromXMLString(content.str());
 
-        for ( auto&& item : transforms)
+    StringBuffer defaultTarget; //This default gives us backward compatibility with only being able to write to the actual request
+    StringBuffer queryName;
+    const char *tgtQueryName = scriptCtx->getXPathString("target/*/@queryname", queryName);
+    defaultTarget.setf("soap:Body/%s/%s", tgtQueryName ? tgtQueryName : method, reqtype);
+
+    for ( auto&& item : transforms)
+    {
+        if (item)
         {
-            if (item)
-            {
-                CEsdlCustomTransform *transform = static_cast<CEsdlCustomTransform*>(item);
-                transform->processTransformImpl(context, theroot, xpathContext, defaultTarget);
-            }
+            CEsdlCustomTransform *transform = static_cast<CEsdlCustomTransform*>(item);
+            transform->processTransformImpl(scriptCtx, theroot, xpathContext, defaultTarget);
         }
+    }
 
-        toXML(theroot, content.clear());
+    scriptCtx->setContent(tgtSection, theroot);
 
-        if (level >= LogMax)
-            DBGLOG(1,"MODIFIED content: %s", content.str());
+    if (level >= LogMax)
+    {
+        scriptCtx->toXML(content.clear(), tgtSection);
```

Comment:
This is helpful, but I found it to be more helpful changing this to the entire document instead of one section. With updates to sections outside of this one possible, seeing the whole document may be more useful.

Reply:
I was emulating what was already logged here, but I can make it the whole blob for now.  That may end up being to verbose, especially as it grows, but we can change it in the future.